src/webgl-surface/webgl-surface.tsx

```diff
@@ -803,6 +825,43 @@ export class WebGLSurface<T extends IWebGLSurfaceProperties, U> extends React.Co
         return response;
       },
 
+      [BaseApplyPropsMethods.IMAGES]: (props: T): IApplyPropsMethodResponse => {
```

Comment:
This needs to be an Async method or return a promise. Because you have a promise inside of it, but you're not returning a promise, you're misleading a caller to believe that the object is fully populated when the result it returned, but it isn't. It's like cake 🍰 . It's a lie.